### PR TITLE
Emitter changes to fuse JS emitting and printing

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

**Comment by gzm0** [comment]
I have added a commit to use `fromByte` in `KnowledgeGuardian` (this change was still in #4917).

File: linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala [hunk 1]
```diff
@@ -575,9 +581,7 @@ final class Emitter(config: Emitter.Config) {
               linkedClass.superClass, // invalidated by class version
               linkedClass.jsSuperClass, // invalidated by class version
               useESClass, // invalidated by class version (depends on kind, config and ancestry only)
-              ctor, // invalidated directly
-              memberMethods, // invalidated directly
-              exportedMembers.flatten // invalidated directly
+              ctor ++ memberMethods ++ exportedMembers.flatten // all 3 invalidated directly
```
Comment (sjrd) [inline linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala:591]: If these are all `List`s, `:::` is more efficient:
```suggestion
              ctor ::: memberMethods ::: exportedMembers.flatten // all 3 invalidated directly
```

**Review by sjrd — APPROVED** [review]
One nit, but otherwise LGTM.

File: linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala (hunk 1 above)
Comment (gzm0) [reply linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala]: Changed .

**Review by sjrd — APPROVED** [review]

**Pull request merged**